R/vismodel.R

```diff
@@ -279,14 +275,14 @@ vismodel <- function(rspecdata,
     illum <- bgil[, grep(illum2, names(bgil))]
   }
   if (illum2 == "ideal") {
-    illum <- rep(1, dim(rspecdata)[1])
+    illum <- rep_len(1, 401)
```

Comment:
Hum, now that I think about it, there is an issue with this since the user could provide a user-defined `S` that has the correct length to match the data but still rely on bundled values for other arguments. In this case, the new `if` branch would never be explored and we would end up the `illum` having the wrong length